extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authentication/BasicHTTPEscalator.java

```diff
@@ -59,6 +60,6 @@ public HttpClient createEscalatedClient(HttpClient baseClient)
   @Override
   public AuthenticationResult createEscalatedAuthenticationResult()
   {
-    return new AuthenticationResult(internalClientUsername, authorizerName, "basic", null);
+    return new AuthenticationResult(internalClientUsername, authorizerName, BASIC, null);
```

Comment:
The "Escalator" is really conceptually linked to one of the "Authenticator" implementations (the authentication used for internal communications has to be something that a configured Authenticator would accept), the interface was split for reasons described here (https://github.com/druid-io/druid/pull/5073)

I guess the way I would handle this now is to add something like a new "druid.escalator.authenticatorName" property that Escalator implementations can use to fill this AuthenticationResult property.

This createEscalatedAuthenticationResult() method is only used on the Broker for some internal SQL code paths (when the broker itself runs a segment metadata query and when it needs to use the SQL planner, see DruidViewMacro), to allow it to reuse existing code paths without larger refactoring or introducing a special "skip authorization checks because I'm the broker" code path. This value doesn't really matter in practice here, and maybe it'd be nice to get rid of the method altogether eventually, so I hadn't commented on it initially.

Reply:
@jon-wei am not sure about adding a new property i don't think it the way to go it adds more confusion then benefits IMO. If i am getting this right there is one to one mapping between escalator type and authenticator name/type. Thus if the user selects/sets the property `druid.escalator.type` it should be enough and the implementation of escalator with type X will set the appropriate name for the authenticator entity. How does this sound?

Reply:
>  If i am getting this right there is one to one mapping between escalator type and authenticator name/type. 

Authenticator name/type are not necessarily correlated (I can name my kerberos authenticator "George" if I wanted).

There's also no restriction on having two Authenticators of the same type (maybe that makes sense for some hypothetical implementation where running two instances with a different "namespace" or similar concept makes sense).

So there is only a mapping between a specific escalator instance and an authenticator instance, which is why I suggest adding the property.

Reply:
hum, i really feel like druid has way too much properties and i feel bad when i add another one to the zillion of properties...let me think more about this

Reply:
@jon-wei since that escalated result is used only by druid and it is not suppose to be involved into the query path coming from a user will set that to `null` i don't see why we need to decorate the proxied query using that result. What you think?

Reply:
I'm okay with having that null for this PR, can you add a comment explaining it (maybe link it to this comment thread)